controllers/gateway/apirule_controller.go

```diff
@@ -118,6 +118,20 @@ func (r *APIRuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		DefaultDomainName: defaultDomainName,
 	}
 
+	convertedApiRule := &gatewayv1beta1.APIRule{}
+	err = r.Client.Get(ctx, req.NamespacedName, convertedApiRule)
+	// If the ApiRule is not found, we don't need to do anything. If it's found and converted, CM reconciliation is not needed.
+	if err == nil && convertedApiRule.Annotations != nil {
```

Comment:
Is there a reason why we don't have a test for this functionality?

Reply:
There is no conversion webhook in unit tests.

Reply:
I don't understand that, what is limiting us? Why do we need a webhook to test this reconciliation?

Reply:
Added the test using annotated APIRule.